hathor/p2p/sync_v2/blockchain_streaming_client.py

```diff
@@ -93,6 +93,15 @@ def handle_blocks(self, blk: Block) -> None:
             self.fails(TooManyVerticesReceivedError())
             return
 
+        # Run basic verification.
+        # if not blk.is_genesis:
+        #     try:
+        #         self.manager.verification_service.validate_basic(blk)
+        #     except TxValidationError as e:
+        #         self.fails(InvalidVertexError(repr(e)))
+        #         return
```

Comment:
This is a TODO, right? Is it expected to be in the final version of this PR?

Reply:
I can't do it right now because I need to refactor the verification services. The issue here is that the chain of blocks is in memory (not in tx storage) so the current verifier will fail with a `TransactionDoesNotExist` error.

Reply:
The refactor I'll do as we agreed [here](https://github.com/HathorNetwork/hathor-core/pull/831#discussion_r1381939630) will potentially solve this, as a list of dependencies will be passed to the verification service, and the storage won't be accessed during verification anymore. Then you can construct this dependency object from memory instead of the storage. I'll work on it today.

Reply:
I guess I'll just leave a TODO note and get back to it later. Is it for you? I'll add this change as a separate task in the project spreadsheet.

Reply:
Ok for me.

Reply:
Done! https://github.com/HathorNetwork/hathor-core/pull/850/commits/93c2fb9b04d747d00a1524e8a2c59016c6a728b3

Reply:
@msbrogli you'll be able to use in-memory transactions for verification using the refactor introduced by these 3 PRs:

- https://github.com/HathorNetwork/hathor-core/pull/831
- https://github.com/HathorNetwork/hathor-core/pull/858
- https://github.com/HathorNetwork/hathor-core/pull/859